hubblestack/extmods/modules/nebula_osquery.py

```diff
@@ -1484,3 +1495,168 @@ def query(query):
+        if extensions_loadfile is None:
+            extensions_loadfile = __opts__.get('osquery_extensions_loadfile')
+
+        autoload = []
+        for f in files:
+            path = f.get('path')
+            dest = f.get('dest')
+            dest = os.path.abspath(dest)
+
+            dest_ok = False
+            whitelisted_paths = __opts__.get('osquery_extensions_path_whitelist',
+                                             ['/opt/osquery/extensions/'])
+            if not isinstance(whitelisted_paths, list):
+                whitelisted_paths = list(whitelisted_paths)
+            for whitelisted_path in whitelisted_paths:
+                if dest.startswith(whitelisted_path):
+                    dest_ok = True
+            if not dest_ok:
+                log.error('Skipping file outside of osquery_extensions_path_whitelist: {0}'
+                          .format(dest))
+                continue
+
+            # Allow for file removals
+            if f.get('remove'):
+                if dest and os.path.exists(dest):
+                    try:
+                        os.unlink(dest)
+                    except:
+                        pass
+                continue
+
+            if not path or not dest:
+                log.error('path or dest missing in files entry: {0}'.format(f))
+                continue
+
+            result = _get_file(**f)
+
+            if result and f.get('extension_autoload', False):
+                autoload.append(dest)
+
+        if extensions_loadfile:
+            try:
+                with open(extensions_loadfile, 'w') as fh:
+                    for extension in autoload:
+                        fh.write(extension)
+                        fh.write('\n')
+            except Exception as exc:
+                log.error('Something went wrong writing osquery extensions.load: {0}'.format(exc))
+
+            # Leave flag to restart osqueryd
+            global OSQUERYD_NEEDS_RESTART
+            OSQUERYD_NEEDS_RESTART = True
+
+
+def _get_file(path, dest, mode='600', user='root', group='root', **kwargs):
+    '''
+    Cache a file from a salt ``path`` to a local ``dest`` with the given
+    attributes.
+    '''
+    try:
+        mode = str(mode)
+        local_path = __salt__['cp.cache_file'](path)
+        if not local_path:
+            log.error('Couldn\'t cache {0} to {1}. This is probably due to '
+                      'an issue finding the file in S3.'.format(path, dest))
+            return False
+        shutil.copyfile(local_path, dest)
+        ret = __salt__['file.check_perms'](name=local_path,
+                                           ret=None,
+                                           user=user,
+                                           group=group,
+                                           mode=mode)
+
+        return ret[0]['result']
+    except Exception as exc:
+        log.error('An error occurred getting file {0}: {1}'.format(path, exc))
```

Comment:
Since most problems have been taken care with getting the file from cache would it make sense to have an error here about copying the file from `local_path` to `dest` (with the exception `exc`)?
We know that `local_path` and `dest` are very likely to exist as far as data values go, so the most likely sources for errors are issues with the action of copying the file. Having both the source and destination paths can be useful context in the error.

Reply:
I don't think it's necessary -- purely because for any given salt:// path you can infer the local path easily by just prepending the cache dir. Additionally, it's theoretically possible that an exception could be thrown before `local_path` is defined so I'd prefer to keep it out of the error message.

Reply:
@fossam Are we good to merge this?